internal/backend/runtime/omni/controllers/omni/internal/exposedservice/exposedservice.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2025 Sidero Labs, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+
+// Package exposedservice provides helpers for controllers to manage ExposedServices.
+package exposedservice
```

Comment:
Moved most of the exposed service logic into this package to keep the controller package sane.